### PR TITLE
HA/DR info

### PR DESCRIPTION
**Pull request by paigehargrave**

### CODE REVIEW

File: modules/ROOT/pages/ha-dr.adoc [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+By default, Runtime Fabric automatically detects and redeploys applications to available resources in the event of a node failure. To best leverage this functionality, Runtime Fabric should have at least one worker node’s worth of unused capacity. With a production configuration of Runtime Fabric, controller nodes can also tolerate a single node failure without impacting availability. 
+
+=== Availability Across Availability Zones
+High-speed, low latency private networks can span multiple availability zones within a region.
+
+Combining the availability across nodes with distributing nodes across different availability zones, Runtime Fabric can be configured to be highly available within a geographical region.
+
+Runtime Fabric is configured according to the type of environment:
+
+* Non-production environments require a minimum of two availability zones for high availability.
```
Comment (sundippatel) [inline modules/ROOT/pages/ha-dr.adoc]: Re-reading this, I think this is misleading given this configuration is not HA to begin with.
We could say for "improved availability".

Comment (paigehargrave) [reply modules/ROOT/pages/ha-dr.adoc]: Updated this - thank you!

File: modules/ROOT/pages/ha-dr.adoc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+* Production environments require a minimum of three availability zones for high availability.
+* DR environments require require a minimum of three availability zones.
```
Comment (sundippatel) [inline modules/ROOT/pages/ha-dr.adoc]: require 2x

Comment (paigehargrave) [reply modules/ROOT/pages/ha-dr.adoc]: Added some text - probably needs some polishing, but wanted to make sure I was on the right track. Thank you!

File: modules/ROOT/pages/ha-dr.adoc [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+The following sections describe HA/DR strategies supported by Runtime Fabric.
+
+== Regions and Availability Zones with Cloud Providers
+
+Cloud providers, such as AWS and Azure, have data centers grouped by geographical regions. Within each region, multiple data centers are classified as Availability Zones. Redundancy can be achieved by creating a private network and using virtual machines across multiple availability zones. Detailed descriptions of regions and availability zones are provided by https://aws.amazon.com/about-aws/global-infrastructure/regions_az/[AWS] and https://docs.microsoft.com/en-us/azure/availability-zones/az-overview[Azure]. 
+
+The number of availability zones is dependent on the region selected for deployment. For high availability, select a region with at least 3 availability zones.
+
+For Runtime Fabric architecture specifications for development and production environments, refer to https://docs.mulesoft.com/runtime-fabric/1.3/architecture.
```
Comment (sundippatel) [inline modules/ROOT/pages/ha-dr.adoc]: [link]Runtime Fabric architecture.

File: modules/ROOT/pages/ha-dr.adoc [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+== High Availability
+
+=== Availability Across Nodes
+By default, Runtime Fabric automatically detects and redeploys applications to available resources in the event of a node failure. To best leverage this functionality, Runtime Fabric should have at least one worker node’s worth of unused capacity. With a production configuration of Runtime Fabric, controller nodes can also tolerate a single node failure without impacting availability. 
+
+=== Availability Across Availability Zones
+High-speed, low latency private networks can span multiple availability zones within a region.
```
Comment (sundippatel) [inline modules/ROOT/pages/ha-dr.adoc]: Preface with "When using a cloud provider like AWS or Azure," ...

Reason being we don't want customers thinking they can do the same with their own, managed data centers.

File: modules/ROOT/pages/ha-dr.adoc [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+Combining the availability across nodes with distributing nodes across different availability zones, Runtime Fabric can be configured to be highly available within a geographical region.
```
Comment (sundippatel) [inline modules/ROOT/pages/ha-dr.adoc]: "When combined with the availability across nodes, Runtime Fabric can be configured to be highly available within a geographical region by distributing nodes across different availability zones."

_Might_ be less "wordy"..

File: modules/ROOT/pages/ha-dr.adoc [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+Runtime Fabric is configured according to the type of environment:
```
Comment (sundippatel) [inline modules/ROOT/pages/ha-dr.adoc]: Runtime Fabric can be configured...

File: modules/ROOT/pages/ha-dr.adoc [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+* Non-production environments require a minimum of two availability zones for high availability.
+* Production environments require a minimum of three availability zones for high availability.
+* DR environments require require a minimum of three availability zones.
+
+For environments with two availability zones, deploy the controller and worker nodes into the two availability zones. You can deploy multiple controllers and workers into specific availability zones.
```
Comment (sundippatel) [inline modules/ROOT/pages/ha-dr.adoc:36]: We should emphasize that this is not a highly available set up (need a min of 3) -- and that another measure is to create a DR RTF.

This little section would benefit with a diagram for each bullet -- showing each availability zone, and the controller and worker nodes that are in each

File: modules/ROOT/pages/ha-dr.adoc (hunk 1 above)
Comment (sundippatel) [reply modules/ROOT/pages/ha-dr.adoc]: Given this isn't actually HA, consider placing it last in the list of bullets.

File: modules/ROOT/pages/ha-dr.adoc [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+Deploy applications and services with two replicas. ( How does this work if we have more replicas than availability zones??? For example, > 3 Replicas. ) In the case of two replicas, if one replica fails, it must be manually restarted to maintain a minimum of 2 active nodes.
```
Comment (sundippatel) [inline modules/ROOT/pages/ha-dr.adoc]: We're confusing replicas with nodes here.

To answer the question -- we can say "each replica should be scheduled to run in a separate node. Note that scheduling of replicas is distributed by different nodes as best-effort, and can vary depending on the availability of each worker node and their available capacity."

Can remove the replicas restarting part -- that's not true and is confusing.

File: modules/ROOT/pages/ha-dr.adoc (hunk 1 above)
Comment (paigehargrave) [reply modules/ROOT/pages/ha-dr.adoc]: Moved this.

File: modules/ROOT/pages/ha-dr.adoc (hunk 3 above)
Comment (paigehargrave) [reply modules/ROOT/pages/ha-dr.adoc]: Added link.

File: modules/ROOT/pages/ha-dr.adoc (hunk 4 above)
Comment (paigehargrave) [reply modules/ROOT/pages/ha-dr.adoc]: Got it - thank you!

File: modules/ROOT/pages/ha-dr.adoc (hunk 5 above)
Comment (paigehargrave) [reply modules/ROOT/pages/ha-dr.adoc]: Yup - agreed. Updated this.

File: modules/ROOT/pages/ha-dr.adoc (hunk 6 above)
Comment (paigehargrave) [reply modules/ROOT/pages/ha-dr.adoc]: Got it - thanks!

File: modules/ROOT/pages/ha-dr.adoc (hunk 7 above)
Comment (paigehargrave) [reply modules/ROOT/pages/ha-dr.adoc:36]: Added a blurb here - please make sure it covers what is needed. Are you aware of any diagrams we can use/modify?

File: modules/ROOT/pages/ha-dr.adoc (hunk 8 above)
Comment (paigehargrave) [reply modules/ROOT/pages/ha-dr.adoc]: Reworded a bit - see if this captures what is needed - thanks!

File: modules/ROOT/pages/ha-dr.adoc (hunk 7 above)
Comment (sundippatel) [reply modules/ROOT/pages/ha-dr.adoc:36]: Looks good. We have some diagrams that we'll likely need to modify to fit these examples.

https://docs.google.com/presentation/d/1yHaO0Sg5X0YwjbqsdXvaA13JYZ9RVTWK6sI6ZmXAzPI/edit?usp=sharing

Comment (paigehargrave) [reply modules/ROOT/pages/ha-dr.adoc:36]: OK - which ones are you thinking? :)

**Comment by IsaacEldridge** [comment]
This PR is over two-years old and is for a version of docs no longer maintained

**Pull request closed without merging**